api/src/main/kotlin/com/gmtkgamejam/plugins/TeamRoutes.kt

```diff
@@ -2,65 +2,91 @@ package com.gmtkgamejam.plugins;
 
 import com.gmtkgamejam.models.Team
 import com.gmtkgamejam.models.TeamCreateDto
+import com.gmtkgamejam.models.TeamDeleteDto
+import com.gmtkgamejam.models.TeamReportDto
+import com.gmtkgamejam.services.TeamService
 import io.ktor.application.*
 import io.ktor.http.*
 import io.ktor.request.*
 import io.ktor.response.*
 import io.ktor.routing.*
 
 fun Application.configureTeamRouting() {
-    val teams = mutableListOf(
-        Team(1, "Dotwo", "1", "Test Team 1", 255, "en", "2021-08-13 14:51:00", "2021-08-13 14:51:00", "2021-08-13 14:51:01",  0),
-        Team(2, "Slam", "2", "Test Team 2", 255, "en", "2021-08-13 14:51:01", "2021-08-13 14:51:01", "",  0),
-        Team(3, "Brysen", "3", "Test Team 3", 255, "en", "2021-08-13 14:51:02", "2021-08-13 14:51:02", "",  0),
-        Team(4, "Dotwo", "1", "Test Team 4", 255, "en", "2021-08-13 14:51:03", "2021-08-13 14:51:03", "",  0),
-    )
+    val service = TeamService()
+
     routing {
         route("/teams") {
             get {
-                call.respond(teams)
+                call.respond(service.getTeams())
             }
             post {
                 val data = call.receive<TeamCreateDto>()
-                val team: Team = Team.fromCreateDto(data)
-                teams.add(team)
-                call.respond(teams)
+                val team = Team.fromCreateDto(data)
+
+                service.createTeam(team)
+                call.respond(team)
             }
             get("{id}") {
-                val id = call.parameters["id"]!!.toInt()
-                call.respond(teams[id - 1])
+                val id = call.parameters["id"]!!.toLong()
+                val team = service.getTeam(id)
+
+                if (team == null) {
+                    call.respondText("Team not found", status = HttpStatusCode.NotFound)
+                }
+
+                call.respond(team!!)
```

Comment:
```suggestion
                team?.let { call.respond(it) }
                call.respondText("Team not found", status = HttpStatusCode.NotFound)
```
Given that the respond calls stop the execution of the following we code we can remove the explicit null check